pkg/client/unversioned/request.go

```diff
@@ -427,8 +427,36 @@ func (r *Request) VersionedParams(obj runtime.Object, convertor runtime.ObjectCo
 		return r
 	}
 	for k, v := range params {
-		for _, vv := range v {
-			r.setParam(k, vv)
+		for _, value := range v {
+			// TODO: Move it to setParam method, once we get rid of
+			// FieldSelectorParam & LabelSelectorParam methods.
+			if k == unversioned.LabelSelectorQueryParam(r.apiVersion) && value == "" {
+				// Don't set an empty selector for backward compatibility.
```

Comment:
I'm not saying that we shouldn't get back to versioned ListOptions.

What I'm saying is that we already had exactly the same code in Request::LabelSelector() and Request::FieldSelector before. So this code wasn't created - it was move from one place to the other.

That said - I agree it shouldn't be here. We should basically move "fieldmappig" to some more generic places (conversion-related) and once this is done, we can also move this code there.
So in my opinion, cleaning up this code is unrelated to getting back to ListOptions...